File: ValidPalindrome/valid_palindrome.js
const isPalindrome = (s) => {
  const san = s.replace(/[^\w]/gi, "").toLowerCase();
  return san.split("").reverse().join("") === san;
};

const pointReversal = (s) => {
  s = s.replace(/[^\w]/gi, "").toLowerCase();
  let left = 0;
  let right = s.length - 1;

  while (left < right) {
    if (s[left] !== s[right]) return false;
    left++;
    right--;
  }

  return true;
};

const run = (() => {
  console.log(isPalindrome("rotators"));
  console.log(pointReversal("rotator"));
})();

File: ValidPalindrome/valid_palindrome.rs
fn main() {
  let palindrome = "rotator";
  let is_palindrome = valid_palindrome(palindrome.to_string());
  println!("{} is palindrome: {}", palindrome, is_palindrome);
}

fn valid_palindrome(s: String) -> bool {
  let s_bytes = s.as_bytes();
  let mut left = 0;
  let mut right = s.len() - 1;

  while left < right {
    if s_bytes[left] != s_bytes[right] {
      return false;
    }

    left += 1;
    right -= 1;
  }

  return true;
}